src/test/optimizer/strategy/between_composition_rule_test.cpp

```diff
@@ -407,4 +407,25 @@ TEST_F(BetweenCompositionTest, NonBoundaryPredicate) {
   EXPECT_LQP_EQ(result_lqp, expected_lqp);
 }
 
+TEST_F(BetweenCompositionTest, NoPullPastDiamond) {
```

Comment:
Unfortunately, I did not find the failing LQP yet because I cannot print TPC-H 7 when it crashes. To spent not too much time on the investigation now, I consider this failing test a part of #1813.